From a824a11026b778bd872eefaf3f4f27a11d0c7527 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Mon, 22 Mar 2010 14:23:45 -0300
Subject: [PATCH 20/29] monitor: convert do_migrate_set_downtime() to QObject

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1269267825-8627-9-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 7971
O-Subject: [PATCH 8/8] monitor: convert do_migrate_set_downtime() to QObject
Bugzilla: 575821
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit c6027f56dc547b5028f0b560bcf4671b6e2a765d)
---
 migration.c     |    3 ++-
 migration.h     |    3 ++-
 qemu-monitor.hx |    3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 migration.c     |    3 ++-
 migration.h     |    3 ++-
 qemu-monitor.hx |    3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/migration.c b/migration.c
index aae4b3f..de32ba7 100644
--- a/migration.c
+++ b/migration.c
@@ -136,7 +136,8 @@ uint64_t migrate_max_downtime(void)
     return max_downtime;
 }
 
-void do_migrate_set_downtime(Monitor *mon, const QDict *qdict)
+void do_migrate_set_downtime(Monitor *mon, const QDict *qdict,
+                             QObject **ret_data)
 {
     double d;
 
diff --git a/migration.h b/migration.h
index 3ac208b..65572c1 100644
--- a/migration.h
+++ b/migration.h
@@ -60,7 +60,8 @@ void do_migrate_set_speed(Monitor *mon, const QDict *qdict, QObject **ret_data);
 
 uint64_t migrate_max_downtime(void);
 
-void do_migrate_set_downtime(Monitor *mon, const QDict *qdict);
+void do_migrate_set_downtime(Monitor *mon, const QDict *qdict,
+                             QObject **ret_data);
 
 void do_info_migrate_print(Monitor *mon, const QObject *data);
 
diff --git a/qemu-monitor.hx b/qemu-monitor.hx
index ad6676e..825caaa 100644
--- a/qemu-monitor.hx
+++ b/qemu-monitor.hx
@@ -778,7 +778,8 @@ ETEXI
         .args_type  = "value:T",
         .params     = "value",
         .help       = "set maximum tolerated downtime (in seconds) for migrations",
-        .mhandler.cmd = do_migrate_set_downtime,
+        .user_print = monitor_user_noop,
+        .mhandler.cmd_new = do_migrate_set_downtime,
     },
 
 STEXI
-- 
1.7.0.3

